### PR TITLE
Test:  remove tools folder from EndToEnd.zip

### LINKED ISSUE #7962: Test:  E2E tests use GenerateTestPackage.exe from \tools folder in EndToEnd.zip used instead of the just built executable
EndToEnd.zip creation involves optional inclusion of a tools folder.  In CI builds the tools folder is always added.  The tools folder includes, among other things, a prebuilt copy of GenerateTestPackages.exe.  A setup script [overwrites](https://github.com/NuGet/NuGet.Client/blob/dev/scripts/e2etests/CopyToolsAndSetupMachine.ps1#L14) the just built GenerateTestPackages.exe with the version in the tools folder.

None of the tools are used anymore.

### PR DESCRIPTION
**Pull request by dtivel**
Resolve https://github.com/NuGet/Home/issues/7962.

None of the tools is used anymore.

This change also uses a prebuilt test data package (NuGet.Client.EndToEnd.TestData) instead of generating test inputs every build.

### CODE REVIEW

**Review by zivkan — APPROVED** [review]

File: build/vsts_build.yaml [hunk 1]
```diff
@@ -255,7 +255,7 @@ phases:
     displayName: "Create EndToEnd Test Package"
     inputs:
       scriptName: "$(Build.Repository.LocalPath)\\scripts\\cibuild\\CreateEndToEndTestPackage.ps1"
-      arguments: "-c $(BuildConfiguration) -tv 15 -out $(Build.Repository.LocalPath)\\artifacts\\VS15 -NuGetToolsPath $(NuGetCIToolsFolder)"
```
Comment (nkolev92) [inline build/vsts_build.yaml:272]: I'd imagine this variable is set somewhere in the ADO variable groups?

**Review by nkolev92 — APPROVED** [review]
👏

File: .nuget/packages.config [hunk 2]
```diff
@@ -1,21 +1,22 @@
...
+  <package id="Microsoft.DotNet.Maestro.Tasks" version="1.1.0-beta.19178.1" />  <!-- for publishing to the .NET Core build asset registry (BAR) -->
   <package id="Microsoft.VisualStudio.Composition" version="15.8.98" targetFramework="net45" />
   <package id="Microsoft.VisualStudio.ProjectSystem" version="16.0.201-pre-g7d366164d0" targetFramework="net45" />
   <package id="Microsoft.VSSDK.BuildTools" version="15.9.3032" targetFramework="net46" />
-  <package id="Newtonsoft.Json" version="9.0.1" targetFramework="net45" />
-  <package id="Lucene.Net" version="3.0.3" targetFramework="net40" />
   <package id="Microsoft.Web.Xdt" version="2.1.2" targetFramework="net40" />
+  <package id="Newtonsoft.Json" version="9.0.1" targetFramework="net45" />
+  <package id="NuGet.Build.Tasks.Pack" version="4.9.2" />
+  <package id="NuGet.Client.EndToEnd.TestData" version="1.0.0" />
```
Comment (dtivel) [inline .nuget/packages.config:15]: This entry is new.  All other changes in this file are just reordering entries by package ID.

**Comment by dtivel** [comment]
@nkolev92, @zivkan, I updated this PR to also use pregenerated E2E test data.  Can you please rereview?

File: build/vsts_build.yaml (hunk 1 above)
Comment (dtivel) [reply build/vsts_build.yaml:272]: Yep, and I'll remove it.

**Review by nkolev92 — APPROVED** [review]

**Pull request merged**